File: scripts/framework/jobs/Cron.lua
Jobs = Jobs or {}

function Jobs.NewCron()
    local self = {}
    local corn = BabyEngine.Cron.New()
    function self.Add( sepc, cb )
        return BabyEngine.Cron.Add(corn, sepc, cb)
    end

    function self.Remove( id )
        return BabyEngine.Cron.Remove(corn, id)
    end

    function self.Stop( id )
        return BabyEngine.Cron.Stop(corn)
    end
    return self
end

File: jobs/cron.go
package jobs

import (
    "github.com/robfig/cron/v3"
)

type Context struct {
    cron *cron.Cron
}

func NewCron() *Context {
    c := &Context{}
    c.cron = cron.New(cron.WithSeconds())
    c.cron.Start()
    return c
}

func (c *Context) Add(spec string, cb func()) (int, error) {

    id, err := c.cron.AddFunc(spec, cb)
    if err != nil {
        return -1, err
    }
    return int(id), nil
}

func (c *Context) Remove(id int) {
    c.cron.Remove(cron.EntryID(id))
}

func (c *Context) Stop()  {
    c.cron.Stop()
}